### DISCUSSION TITLE
Trying to load payloads on Maya USD proxy nodes not working after have been previously disabled

### DISCUSSION THREAD

**Discussion by stephanehoarau** [Autodesk/maya-usd#3351] [General]
Hello !
We hit an issue with the Load Payloads option (either in UI or by script) on USD proxy nodes.
I created a simple `demo_example.zip` file composed of two files to expose you the problem and the steps to reproduce the issue.
- example.usd (ascii USD file)
- geo_payload.usd (binary USD file)
The `example.usd` will contain a default prim called "main" that is composed of a payload file (geo_payload.usd). 

**Steps to reproduce the issue:**
```
    - Load Maya (2023 + USD 0.22.0 plugin)
    - Create a stage with the menu "Edit > Universal Scene Description (USD) > Stage with New Layer" -> the stage node "stage1" is created within an "anonymousLayer1" USD layer.
    - Launch the "USD Layer Editor" by RMB on the "stage1" node in the Outliner.
    - In the "USD Layer Editor", select the option "Load SubLayers..." with the RMB on the "anonymousLayer1*" layer.
    - In the new UI "Load SubLayers in the anonymousLayer1", choose the "Layer Path" option and select in your Browser the provided file "example.usd" .
    - This will show you a wonderfull sphere on your scene.
    - Then select the "stageStage1" shape related to "stage1".
    - in the "Attribute Editor", in the "Extra Attributes" section, find the checkbox called "Load Payloads":
        - If you untick the checkbox, you should see the sphere disappear in the viewport. Actually this is the expected behavior.
        - If you tick again the checkbox, you should see the sphere appear in the viewport. This is the expected behavior.

    - Now save this scene in a maya file (with Save Scene Options: Save all edits (including USD) to the Maya scene file (I saved it under "[stage_example.ma]")
    - Clean the scene with File > New.
    - Load the scene just saved.
    - Select the "stageStage1" shape in the scene. 
    - in the "Attribute Editor", in the "Extra Attributes" section, find the checkbox called "Load Payloads": 
        - If you untick the checkbox, you should see the sphere disappear in the viewport. This is the expected behavior.
        - Now if you tick again the checkbox, unfortunately the sphere won't appear again. THIS IS NOT EXPECTED. Moreover, notice also that the sublayer called "example.usd" under "anonymousLayer1" has been removed too (we do believe this is actually the main issue though).
```

[demo_example.zip](https://github.com/Autodesk/maya-usd/files/12742884/demo_example.zip)

**Comment by sharktacos** [comment]
I believe the solution would be to save the USD stage, as "anonymous Layer" means the USD layer only exists temporarily in memory.

**Comment by santosd** [comment]
I ran some tests on this and It seems that either way the issue seems to be broken. Whether I save the USD to the Maya file or save the Anonymous layer to its own USD file. I have issues with the Load Payloads option. I see your issue when I save everything to the Maya file. When I save things separately I see a different issue where the prims are always visible regardless of Load Payloads being checked or unchecked. I will log this issue internally and bring it over to our issues.

**Comment by santosd** [comment]
Created this issue from this discussion: https://github.com/Autodesk/maya-usd/issues/3362

**Comment by womanyoyoyo-adsk** [comment]
Hi @stephanehoarau! For the 'Load' action here, would you expect your payloads to load with descendants or to not load the descendants?

**Reply by stephanehoarau** [reply]
I think the two options would be interesting, but I would suggest that the default to load without descendants, because it could take time.
And the second option with the descendants would warn about a long loading time.

**Comment by santosd** [comment]
@stephanehoarau another thing to make note of is that these things can be loaded and unloaded from the Outliner through the right click menu.
![image](https://github.com/Autodesk/maya-usd/assets/29235905/c10ee95e-553b-4ff8-8a4b-7b8680d64dc2)
![image](https://github.com/Autodesk/maya-usd/assets/29235905/d1b4eb9e-7942-4609-9753-2379a2efa146)
